File: 240605/키를 기준으로 정렬/sort-by-height.swift
import Foundation

precedencegroup ForwardPipe {
    associativity: left
}

infix operator |> : ForwardPipe

func |> <T, U> (value: T, function: (T) -> U) -> U {
    return function(value)
}

class Person: Hashable, Comparable, CustomStringConvertible {
    let name: String
    let tall: Int
    let weight: Int
    
    init(name: String, tall: Int, weight: Int) {
        self.name = name
        self.tall = tall
        self.weight = weight
    }

    func hash(into hasher: inout Hasher) {
        hasher.combine(name)
        hasher.combine(tall)
        hasher.combine(weight)
    }

    static func == (lhs: Person, rhs: Person) -> Bool {
        return lhs.name == rhs.name
                && lhs.tall == rhs.tall
                && lhs.weight == rhs.weight
    }

    static func < (lhs: Person, rhs: Person) -> Bool {
        return lhs.tall < rhs.tall
    }

    var description: String {
        return "\(name) \(tall) \(weight)"
    }
}

func readNum() -> Int? {
    guard let line = readLine(), let num = Int(line) else { return nil }
    return num
}

func readPerson() -> Person? {
    guard let line = readLine() else { return nil }
    let temp = line.split(separator: " ")
    guard temp.count == 3,
        let tall = Int(temp[1]),
        let weight = Int(temp[2]) else { return nil }
    
    return Person(name: String(temp[0]), tall: tall, weight: weight)
}

func solution(people: [Person]) -> String {
    return people.sorted().map{$0.description}.joined(separator: "\n")
}

func main() {
    guard let n = readNum() else { return }
    let people = (1...n).compactMap({_ in readPerson()})
    print(people |> solution)
}

main()

File: 250214/키를 기준으로 정렬/sort-by-height.py
n = int(input())
name = []
height = []
weight = []

for _ in range(n):
    n_i, h_i, w_i = input().split()
    name.append(n_i)
    height.append(int(h_i))
    weight.append(int(w_i))

# Write your code here!

class Person:

    def __init__(self, name, height, weight):
        self.name = name
        self.height = height
        self.weight = weight

    def __str__(self):
        return f"{self.name} {self.height} {self.weight}"


people = []

for i in range(n):
    people.append(Person(name[i], height[i], weight[i]))

people.sort(lambda x: x.height)

for person in people:
    print(person)